course_discovery/apps/publisher/views.py

```
@@ -362,13 +362,40 @@ class CourseEditView(mixins.PublisherPermissionMixin, UpdateView):
     """ Course Edit View."""
     model = Course
     form_class = CourseForm
+    entitlement_form = CourseEntitlementForm
     permission = OrganizationExtension.EDIT_COURSE
     template_name = 'publisher/course_edit_form.html'
     success_url = 'publisher:publisher_course_detail'
 
     def get_success_url(self):
         return reverse(self.success_url, kwargs={'pk': self.object.id})
 
+    def get_context_data(self, **kwargs):
+        context = super(CourseEditView, self).get_context_data(**kwargs)
+        history_id = self.request.GET.get('history_id', None)
+        self.object = self.get_object()
+
+        try:
+            history_object = self.object.history.get(history_id=history_id) if history_id else None
+        except:  # pylint: disable=bare-except
```

Bare excepts should be exceedingly rare. At the least, they should be "except Exception:" because otherwise you're going to catch KeyboardInterrupt and SystemExit exceptions.

But here, maybe just catch DoesNotExist? (unless you want to worry about multiple rows too)

---

Not my code sadly though I can definitely update this, not positive what errors are expected here.

---

Ah, I hadn't read far enough to see that this was a copy paste job. Can you at least update it to "except Exception" then? Bare excepts hurt my soul.

---

Bare except vs Broad except, but you're right it's the smallest amount better :(

---

My distaste for bare excepts are more about quality than quantity. The kind of things that a bare except catches just shouldn't be caught. You don't want this try/except to intercept a Ctrl+C (KeyboardInterrupt), and you don't want it to intercept a sys.exit() (SystemExit).

---

Python lets you catch some surprising things. :-/